File: backend/app/schemas/resume.py
from datetime import datetime
from typing import Optional

from pydantic import BaseModel


class ResumeGenerateRequest(BaseModel):
    application_id: int


class AtsAuditResult(BaseModel):
    score: int
    matched_keywords: list[str] = []
    missing_keywords: list[str] = []
    formatting_issues: list[str] = []
    suggestions: list[str] = []


class GapAnalysisResult(BaseModel):
    matching_skills: list[str] = []
    missing_skills: list[str] = []
    strengths: list[str] = []
    recommendations: list[str] = []


class ResumeResponse(BaseModel):
    id: int
    application_id: int
    version: int
    resume_json: dict
    pdf_url: Optional[str]
    ats_score: Optional[int]
    ats_audit_result: Optional[dict]
    gap_analysis: Optional[dict]
    created_at: datetime

    model_config = {"from_attributes": True}


class CoverLetterGenerateRequest(BaseModel):
    application_id: int


class CoverLetterResponse(BaseModel):
    id: int
    application_id: int
    version: int
    content: str
    created_at: datetime

    model_config = {"from_attributes": True}

File: frontend/src/types/resume.ts
export interface ResumeContact {
  name?: string;
  email?: string;
  phone?: string;
  location?: string;
  linkedin?: string;
}

export interface ResumeExperience {
  title: string;
  company: string;
  location?: string;
  start?: string;
  end?: string;
  bullets: string[];
}

export interface ResumeEducation {
  degree: string;
  institution: string;
  year?: string;
}

export interface ResumeJson {
  contact?: ResumeContact;
  summary?: string;
  experience?: ResumeExperience[];
  education?: ResumeEducation[];
  skills?: string[];
  certifications?: string[];
}

export interface AtsAuditResult {
  score: number;
  matched_keywords: string[];
  missing_keywords: string[];
  formatting_issues: string[];
  suggestions: string[];
}

export interface GapAnalysisResult {
  matching_skills: string[];
  missing_skills: string[];
  strengths: string[];
  recommendations: string[];
}

export interface Resume {
  id: number;
  job_id: number;
  version: number;
  resume_json: ResumeJson;
  pdf_url: string | null;
  ats_score: number | null;
  ats_audit_result: AtsAuditResult | null;
  gap_analysis: GapAnalysisResult | null;
  created_at: string;
}

export interface CoverLetter {
  id: number;
  job_id: number;
  version: number;
  content: string;
  created_at: string;
}
